packages/salesforcedx-vscode-apex/src/languageServer.ts

```diff
@@ -55,6 +55,37 @@ const createServer = async (extensionContext: vscode.ExtensionContext): Promise<
       .getConfiguration()
       .get<boolean>('salesforcedx-vscode-apex.advanced.enable-completion-statistics', false);
 
+    // Configurations of the definitions of eligible apex classes/methods/properties
```

Comment:
Look into passing these values as part of LanguageClientOptions section initializationOptions, just like you did for feature enableSynchronizedInitJobs.